nomos-services/cryptarchia-consensus/src/states.rs

```diff
@@ -0,0 +1,132 @@
+            _network_adapter_settings: Default::default(),
+            _blend_adapter_settings: Default::default(),
+        }
+    }
+
+    pub(crate) fn from_cryptarchia(cryptarchia: &Cryptarchia, leader: &Leader) -> Self {
+        let security_block_header = cryptarchia.consensus.get_security_block_header_id();
+        let security_ledger_state = security_block_header
+            .and_then(|header| cryptarchia.ledger.state(&header))
+            .cloned();
+        let security_leader_notes = security_block_header
+            .and_then(|header_id| leader.notes(&header_id))
+            .cloned();
+
+        Self::new(
+            Some(cryptarchia.tip()),
+            security_block_header,
+            security_ledger_state,
+            security_leader_notes,
+        )
+    }
+
+    fn can_recover(&self) -> bool {
+        // This only checks whether tip is defined, as that's a state variable that should
+        // always exist. Other attributes might not be present.
+        self.tip.is_some()
+    }
+
+    fn can_recover_from_security(&self) -> bool {
+        // TODO: Check if one or more (but not all) the security attrs are missing. That's a bug.
+        self.can_recover()
+            && self.security_block.is_some()
+            && self.security_ledger_state.is_some()
+            && self.security_leader_notes.is_some()
+    }
+
+    pub fn recovery_strategy(&mut self) -> CryptarchiaInitialisationStrategy {
+        if self.can_recover_from_security() {
+            let strategy = SecurityRecoveryStrategy {
+                tip: self.tip.take().expect("tip not available"),
+                security_block_id: self
+                    .security_block
+                    .take()
+                    .expect("security block not available"),
+                security_ledger_state: self
+                    .security_ledger_state
+                    .take()
+                    .expect("security ledger state not available"),
+                security_leader_notes: self
+                    .security_leader_notes
+                    .take()
+                    .expect("security leader notes not available"),
+            };
+            CryptarchiaInitialisationStrategy::RecoveryFromSecurity(Box::new(strategy))
+        } else if self.can_recover() {
+            let strategy = GenesisRecoveryStrategy {
+                tip: self.tip.expect("tip not available"),
+            };
+            CryptarchiaInitialisationStrategy::RecoveryFromGenesis(strategy)
+        } else {
+            CryptarchiaInitialisationStrategy::Genesis
+        }
+    }
+}
+
+impl<TxS, BxS, NetworkAdapterSettings, BlendAdapterSettings> ServiceState
+    for CryptarchiaConsensusState<TxS, BxS, NetworkAdapterSettings, BlendAdapterSettings>
+{
+    type Settings = CryptarchiaSettings<TxS, BxS, NetworkAdapterSettings, BlendAdapterSettings>;
+    type Error = Error;
+
+    fn from_settings(_settings: &Self::Settings) -> Result<Self, Self::Error> {
```

Comment:
Nothing to do with this PR, but this is the bit I was talking about elsewhere. All states must do this, and it looks like every time the implementation will always be the same, which to me indicates it's not optimal design.

Reply:
The idea behind this iirc is that you may overwrite saved state from settings on some services.

Reply:
But state should not contain settings, right? Those are always fetched when the service is started or within the service's `run` loop, so I don't see a case in which a state might do that.

Reply:
Cause all I've seen is for the service to have settings for the different components, but having settings for a service's state, to me, sounds less than ideal. But maybe there is a concrete use case in which this is needed, not sure.

Reply:
> Cause all I've seen is for the service to have settings for the different components, but having settings for a service's state, to me, sounds less than ideal. But maybe there is a concrete use case in which this is needed, not sure.

It is not settings for the service state. It is the settings for the service itself. Its true that it could be done from within the service. Meaning that a service would recover its saved state, it will have the settings and to whatever it need to do from it. Instead of delegating it to the state operator.

Reply:
Yeah but the trait defines a `type Settings` for the state, which effectively makes it state's settings. There are then bounds that bound it to the service's state, but that is not part of the trait definition. As for your second sentences, I am not sure I understand if you agree with me, or making the case for having this solution? Let's say some stuff must be recovered from a remote DB which requires a bunch of configs. Wouldn't that all be part of the recovery operator, which would be configured with some of the service's settings? 

What I am arguing here is that having this `type Settings` and `fn from_settings` for a service state is not necessary, but perhaps you can help me better understand your previous point?

Reply:
> As for your second sentences, I am not sure I understand if you agree with me, or making the case for having this solution?

I partially agree with you. Partially making the case. 
Usually, we imagine the state as the composition of some `settings` + `state_N`. At the start, it would just be a state from settings. Later on, it would be the `state_13` + `settings`. It may not make sense for all services *now*, but, it was how it was conceived. Now, *this* could be done by the state operator or by the service itself. We decided to delegate that to the operator.
I see your point about removing it as it is not used atm. But I think it may as well just be kept here. It doesn't harm having it. Unless you have a better proposition than just removing it.

Reply:
> What I am arguing here is that having this type Settings and fn from_settings for a service state is not necessary, but perhaps you can help me better understand your previous point?

It is true that maybe this should be the `ServiceSettings` itself :thinking: